app/supplier_utils.py

```diff
@@ -37,3 +44,80 @@ def check_supplier_role(role, supplier_id):
         abort(400, "'supplierId' is required for users with 'supplier' role")
     elif role != 'supplier' and supplier_id:
         abort(400, "'supplierId' is only valid for users with 'supplier' role, not '{}'".format(role))
+
+
+def company_details_confirmed_if_required_for_framework(framework_slug, supplier_framework):
+    return (
+        supplier_framework.supplier.company_details_confirmed
+        if framework_slug == 'g-cloud-10' else
+        supplier_framework.application_company_details_confirmed
+    )
+
+
+def update_open_declarations_with_company_details(db, supplier_id, updater_json, specific_framework_slug=None):
+    """Expected to be called within a view"""
+    open_supplier_frameworks_query = SupplierFramework.query.filter(
```

Comment:
Could rename this var `supplier_frameworks_query` to be clearer. Or could add the check for frameworks being open here and then not worry about the `else` below.

Reply:
Kept it as-is after moving the `open` filter up to here.